docs/compatibility.md

```diff
@@ -352,4 +352,10 @@ Casting from string to timestamp currently has the following limitations.
 - <a name="Footnote1"></a>[1] The timestamp portion must be complete in terms of hours, minutes, seconds, and
  milliseconds, with 2 digits each for hours, minutes, and seconds, and 6 digits for milliseconds. 
  Only timezone 'Z' (UTC) is supported. Casting unsupported formats will result in null values. 
- 
+
+### Constant Folding
+
+ConstantFolding is an operator optimization rule in Catalyst that replaces expressions that can
+be statically evaluated with their equivalent literal values. We don't support excluding the rule 
+`org.apache.spark.sql.catalyst.optimizer.ConstantFolding` as we rely on this rule internally 
```

Comment:
Rather than just squirreling this away in the corner of a doc most users will not read, why not also update the plugin to check this config setting when it starts to optimize a plan?  It could emit a warning and refuse to optimize the plan which will let the query run correctly, albeit completely on the CPU.

Reply:
The issue that would happen is that we get two literal values in say a GPU Add and we try to run it and it blows up. I agree that we probably want to disable the plugin, but at least output a warning if we see that ConstantFolding has been disabled.

As an alternative when replacing the plan with a GPU enabled version we could look for all foldable CPU operators, and if they are not on Literal values we CANNOT replace them.  Because then at least we could not crash, and we would honor what was asked by the end user, to disable constant folding.